From 0152e30450e1ae5444163e68adeaef9b5590d777 Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Thu, 8 Feb 2018 12:48:01 +0200
Subject: [PATCH 3353/5242] MLK-17528-2: ASoC: imx_pdm: Use FSL_SAI_CLK_BIT to
 signal the proper clock id

commit  989d39aa824bf89de87b25a9b30a44760439c52e from
https://source.codeaurora.org/external/imx/linux-imx.git

The current implementation suggest that MAST1 frequency is to be changed,
which is wrong. Use FSL_SAI_CLK_BIT clock id instead of FSL_SAI_CLK_MAST1
in order to make the code more intuitive and to signal proper
clk_id to SAI.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-pdm.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/fsl/imx-pdm.c b/sound/soc/fsl/imx-pdm.c
index eea0830..e8cea52 100644
--- a/sound/soc/fsl/imx-pdm.c
+++ b/sound/soc/fsl/imx-pdm.c
@@ -80,7 +80,7 @@ static int imx_pdm_mic_hw_params(struct snd_pcm_substream *substream,
 	/* set tdm slots only one for now */
 	snd_soc_dai_set_tdm_slot(cpu_dai, 0, 0, 1, 32);
 	/* Set clock out */
-	ret = snd_soc_dai_set_sysclk(cpu_dai, FSL_SAI_CLK_MAST1,
+	ret = snd_soc_dai_set_sysclk(cpu_dai, FSL_SAI_CLK_BIT,
 			bitclk, SND_SOC_CLOCK_OUT);
 	if (ret) {
 		dev_err(card->dev, "fail to set cpu sysclk: %d\n", ret);
-- 
1.7.9.5

